sycl/source/detail/event_impl.hpp

```diff
@@ -164,14 +164,14 @@ class event_impl {
   /// Returns vector of event dependencies.
   ///
   /// @return a reference to MPreparedDepsEvents.
-  std::vector<std::shared_ptr<event_impl>> &getPreparedDepsEvents() {
+  std::vector<std::weak_ptr<event_impl>> &getPreparedDepsEvents() {
```

Comment:
Could you please tell who own/holds `event_impl` with this patch?

Still think that approach mentioned in this comment would be much simpler https://github.com/intel/llvm/pull/4793#discussion_r737389632.